services/data-submission.js

```diff
@@ -0,0 +1,131 @@
+const { NEW, IN_PROGRESS, SUBMITTED, RELEASED, COMPLETED, ARCHIVED} = require("../constants/data-submission-constants");
+const {DATA_SUBMISSIONS_COLLECTION: DATA_SUBMISSION} = require("../crdc-datahub-database-drivers/database-constants");
+const {v4} = require('uuid')
+const {getCurrentTime, subtractDaysFromNow} = require("../crdc-datahub-database-drivers/utility/time-utility");
+const {HistoryEventBuilder} = require("../domain/history-event");
+const {verifySession} = require("../verifier/user-info-verifier");
+const ERROR = require("../constants/error-constants");
+const {getSortDirection} = require("../crdc-datahub-database-drivers/utility/mongodb-utility");
+const USER_CONSTANTS = require("../crdc-datahub-database-drivers/constants/user-constants");
+const {USER} = require("../crdc-datahub-database-drivers/constants/user-constants");
+const ROLES = USER_CONSTANTS.USER.ROLES;
+
+class DataSubmission {
+    constructor(logCollection, dataSubmissionCollection, organizationService, userService, dbService, notificationsService, emailParams) {
+        this.logCollection = logCollection;
+        this.dataSubmissionCollection = dataSubmissionCollection;
+        this.organizationService = organizationService;
+        this.userService = userService;
+        this.dbService = dbService;
+        this.notificationService = notificationsService;
+        this.emailParams = emailParams;
+    }
+
+    async createDataSubmission(params, context) {
+        verifySession(context)
+            .verifyInitialized()
+            .verifyRole([ROLES.SUBMITTER, ROLES.ORG_OWNER]);
+        let dataSubmission=params.dataSubmission;
+        let userInfo = context.userInfo;
+        let newApplicationProperties = {
+            _id: v4(undefined, undefined, undefined),
+            name: params.name,
+            submitterID: userInfo._id,
+            submitterName: formatApplicantName(userInfo),
+            organization: {_id:"testid", name: userInfo.organization.orgName},
+            dataCommons: "CDS",
+            modelVersion: "string for future use",
+            studyAbbreviation: params.studyAbbreviation,
+            dbGapID: params.dbGapID,
+            bucketName: "todo: get from database",
+            rootPath: "todo: organization/study?",
+            status: NEW,
+            history: [HistoryEventBuilder.createEvent(userInfo._id, NEW, null)],
+            concierge: "todo: get from database?",
```

Comment:
Could you please use a "// TODO" comment instead? The IDE will identify and track these TODO tasks.